externals/kyuubi-spark-sql-engine/src/test/scala/org/apache/kyuubi/engine/spark/schema/RowSetBenchmark.scala

```diff
@@ -0,0 +1,95 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.kyuubi.engine.spark.schema
+
+import scala.math.BigDecimal.RoundingMode
+
+import org.apache.commons.lang3.time.StopWatch
+import org.apache.hive.service.rpc.thrift.TProtocolVersion
+import org.apache.spark.sql.Row
+import org.apache.spark.sql.types._
+
+class RowSetBenchmark extends BaseRowSetSuite {
+  test("to row set benchmark") {
```

Comment:
don't put benchmarks into tests

Reply:
Would like to keep it as running via tests by setting `RUN_BENCHMARK=1`, just like other existed benchmarks like TPCDSTableGenerateBenchmark.

JMH for isolated benchmark testing could be introduced next time.
Having trouble in integrating JMH for Scala without official Maven plugin support , using JMH Java annotations , the  proper execution entry point to run with JMH and the isolation path for JMH benchmarks.

Reply:
Hi @yaooqinn . I have changed to use Spark's Benchmark tools for running and generating the benchmarks, just like the `TPCDSTableGenerateBenchmark` and `ZorderCoreBenchmark`. Could you have a look? 
And thanks for the guidance from @wForget .

Reply:
-1 to use Spark's Benchmark tools in engine modules

Reply:
We could refactor this onto JMH in the follow-up PRs. These tests are not run with GA tests. This should not be a blocker issue here for evaluating the overall TRowSet generation.

Reply:
> -1 to use Spark's Benchmark tools in engine modules

what's the reason/major concern

Reply:
> We could refactor this onto JMH in the follow-up PRs

We don't need to refactor if it's originally designed with JMH

Reply:
[jmh-scala-benchmark-archetype](https://mvnrepository.com/artifact/org.openjdk.jmh/jmh-scala-benchmark-archetype)